### ISSUE TITLE
[ISSUE] different databricks_grants inside different modules overwrite eachother

### ISSUE THREAD

**Issue by jacovg91** [databricks/terraform-provider-databricks#2704]
Hello,

I have a multi module setup and I'm applying multiple databricks_grants resource blocks (one or more in each module) 

### Configuration

Module A: 

```hcl
resource "databricks_grants" "use_catalog" {
  provider  = databricks.workspace
  catalog   = "strada"
  grant {
    principal  = "Group 1
    privileges = ["USE_CATALOG"]
  }
}
```

Module B:
```hcl
resource "databricks_grants" "use_catalog" {
  provider  = databricks.workspace
  catalog   = "strada"
  grant {
    principal  = "Group 2"
    privileges = ["USE_CATALOG"]
  }
}
```

### Expected Behavior
In this case, databricks_grants.use_catalog in Module A and databricks_grants.use_catalog in Module B have different addresses due to the module prefix. This should ensure that they are treated as separate resources and will not overwrite each other during the apply process.

### Actual Behavior
Every `terraform apply` i run, they keep changing from group 1 to group 2.

**Comment by alexott** [comment]
this is related to #2395

**Comment by jacovg91** [comment]
Even if i give them a different name in the different modules, i.e. use_catalog_a, and use_catalog_b, they will still overwrite eachother each apply. 

This is extremely problematic and blocks us from properly implementing grants in our organization (which contains over 80 Databricks workspaces with their own AD groups). Is there no workaround for this? I guess the only way is just using one `databricks_grants` resource block and putting everything in there.

**Comment by alexott** [comment]
no workaround until #2395 is implemented. The current behavior is [documented](https://registry.terraform.io/providers/databricks/databricks/latest/docs/resources/grants):

> It is required to define all permissions for a securable in a single resource, otherwise, Terraform cannot guarantee config drift prevention.

**Comment by alexott** [comment]
yes, right now all permissions for a single object should be defined in a single resource. For another object - it will be another resource.

**Comment by jacovg91** [comment]
> yes, right now all permissions for a single object should be defined in a single resource. For another object - it will be another resource.

Thanks a bunch Alex. It'll do for a workaround, but it doesn't make the code very clean, no `for_each `loops for example :(

**Comment by alexott** [comment]
You can use dynamic blocks

**Comment by alexott** [comment]
Will be fixed by #3024

**Comment by jacovg91** [comment]
> Will be fixed by #3024

Great news @alexott, thanks for the notification!
